finagle-thrift/src/test/scala/com/twitter/finagle/thrift/EndToEndTest.scala

```diff
@@ -519,6 +520,32 @@ class EndToEndTest extends FunSuite with ThriftTest with BeforeAndAfter {
 
     Await.ready(ss.close())
   }
+
+  test("clientId is not sent and prep stats are not recorded when TTwitter upgrading is disabled") {
```

Comment:
this is sufficient I think but if you're feeling motivated, the upgrade request will show up in tracing as well so we could verify that disabling upgrades yields 1 fewer request trace.